File: Cpp/Path Sum.cpp
#include<iostream>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};


bool hasPathSum(TreeNode *root, int sum)
    {
        if(root==NULL&&sum!=0) return false;
        else if(root->left==NULL&&root->right==NULL&&root->val==sum) return true;
        else return hasPathSum(root->left, sum-root->val)||hasPathSum(root->right, sum-root->val);
    }

File: Python/Path Sum.py
#!/usr/bin/env python

# Definition for a  binary tree node
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


def hasPathSum(self, root, sum):
    if root == None:
        return False
    if root.left == None and root.right == None:
        if root.val == sum:
            return True
        else:
            return False
    return self.hasPathSum(root.left, sum - root.val) or self.hasPathSum(root.right, sum - root.val)